src/buildstream/_project.py

```diff
@@ -382,12 +402,18 @@ def alias_exists(self, alias, *, first_pass=False):
     # get_alias_uris()
     #
     # Args:
-    #    alias (str): The alias.
-    #    first_pass (bool): Whether to use first pass configuration (for junctions)
-    #    tracking (bool): Whether we want the aliases for tracking (otherwise assume fetching)
+    #    alias: The alias.
+    #    first_pass: Whether to use first pass configuration (for junctions)
+    #    tracking: Whether we want the aliases for tracking (otherwise assume fetching)
+    #
+    # Returns:
+    #    A list of (SourceMirror, string) tuples with each alias substitution found along
+    #    with the SourceMirror object which produced it.
```

Comment:
As far as I can tell, this returns a list with elements that can be strings (alias substitution for the legacy/default implementation) or mirror objects, not a list of tuples. These lists/values of imprecise type are passed through various functions that expect a string (according to the parameter type). Is this something you're still working on or am I misreading the code?

Reply:
Indeed. I changed the code and forgot to change the comment.

Reply:
> These lists/values of imprecise type are passed through various functions that expect a string (according to the parameter type)

Forgot to change the type annotations.

Reply:
Updated.